server/src/main/java/org/elasticsearch/action/admin/cluster/stats/DenseVectorFieldStats.java

```diff
@@ -31,28 +37,49 @@ public final class DenseVectorFieldStats extends FieldStats {
         indexedVectorCount = 0;
         indexedVectorDimMin = UNSET;
         indexedVectorDimMax = UNSET;
-    }
-
-    DenseVectorFieldStats(StreamInput in) throws IOException {
-        super(in);
-        indexedVectorCount = in.readVInt();
-        indexedVectorDimMin = in.readVInt();
-        indexedVectorDimMax = in.readVInt();
+        vectorIndexTypeCount = new HashMap<>();
+        vectorSimilarityTypeCount = new HashMap<>();
+        vectorElementTypeCount = new HashMap<>();
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
+        assert false : "writeTo should not be called on DenseVectorFieldStats";
         super.writeTo(out);
         out.writeVInt(indexedVectorCount);
         out.writeVInt(indexedVectorDimMin);
         out.writeVInt(indexedVectorDimMax);
+        if (out.getTransportVersion().onOrAfter(DENSE_VECTOR_INDEX_TYPE_TELEMETRY)) {
+            out.writeMap(vectorIndexTypeCount, StreamOutput::writeString, StreamOutput::writeVInt);
```

Comment:
should we write other maps as well: `vectorSimilarityTypeCount`, `vectorElementTypeCount`?

Reply:
I am thinking of removing the serialization code altogether. It ain't used and shouldn't be. Any internode serialization would have failed regardless because the appropriate deserialization code is never used.